SectionedCollectionView/Views/SectionedCollectionView.swift

```diff
@@ -168,133 +151,55 @@ class SectionedCollectionView: UIView {
-            cell.configure(withValue: item)
-            return cell
-        })
-        
-        self.dataSource.configureSupplementaryView = {(dataSource, collectionView, kind, indexPath) -> UICollectionReusableView in
-            if (kind == UICollectionElementKindSectionHeader) {
-                let header = collectionView.dequeueReusableSupplementaryView(ofKind: UICollectionElementKindSectionHeader, withReuseIdentifier: self.settings.viewCells.headerViewCellReuseIdentifier, for: indexPath) as! ItemCollectionViewCell
-                header.configure(withValue: dataSource[indexPath.section])
-                return header
-            } else {
-                let footer = collectionView.dequeueReusableSupplementaryView(ofKind: UICollectionElementKindSectionFooter, withReuseIdentifier: self.settings.viewCells.footerViewCellReuseIdentifier, for: indexPath) as! ItemCollectionViewCell
-                footer.configure(withValue: dataSource[indexPath.section])
-                return footer
-            }
-        }
+        collectionView.dataSource = self
+        collectionView.delegate = self
     }
     
-    fileprivate func setupBindings() {
-        _sections.asObservable()
-            .bind(to: sections)
-            .disposed(by: disposeBag)
-        
-        self.sections.asObservable()
-            .bind(to: self.collectionView.rx.items(dataSource: self.dataSource))
-            .disposed(by: disposeBag)
-        
-        collectionView.rx.itemSelected
-            .subscribe(onNext: { indexPath in
-                let selectedItemsCount = self.sections.value.compactMap({ sections -> [CustomData] in
-                    return sections.selectedItems().items
-                }).flatMap({ $0 }).count
-                
-                if (self.sections.value[indexPath.section].items[indexPath.row].selected || self.settings.data.selectedLimit == nil || selectedItemsCount < self.settings.data.selectedLimit!) {
-                    self.sections.value[indexPath.section].items[indexPath.row].selected = !self.sections.value[indexPath.section].items[indexPath.row].selected
-                } else {
-                    self._limitReached.asObserver().onNext(())
-                }
-            })
-            .disposed(by: disposeBag)
-    }
-    
-    // MARK: - Public override variables
-    
-    var headerReferenceSize: CGSize {
-        
-        set {
-            let collectionViewFlowLayout = collectionView.collectionViewLayout as? UICollectionViewFlowLayout
-            collectionViewFlowLayout?.headerReferenceSize = newValue
-        }
-        
-        get {
-            let collectionViewFlowLayout = collectionView.collectionViewLayout as? UICollectionViewFlowLayout
-            return collectionViewFlowLayout?.headerReferenceSize ?? .zero
+}
+
+extension SectionedCollectionView: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
```

Comment:
Add an empty line here and one at the bottom of the extension if there isn't already